File: class3e/app/src/main/java/io/github/lee_seul/class3e/MainActivity.java
package io.github.lee_seul.class3e;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button = (Button) findViewById(R.id.btn_call);
        button.setOnClickListener(this);
    }


    @Override
    public void onClick(View v) {
        //Toast.makeText(this, "cliick!", Toast.LENGTH_SHORT).show();

//        Intent intent = new Intent();
//        intent.setAction(Intent.ACTION_DIAL);
//        startActivity(intent);

        Intent intent = new Intent(MainActivity.this, CallActivity.class);
        intent.putExtra("intent-message", "game joa joa");
        startActivity(intent);

    }
}

File: Flashligth/app/src/main/java/io/github/lee_seul/flashligth/MainActivity.kt
package io.github.lee_seul.flashligth

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val intent = Intent(this, TorchService::class.java)

        flashSwich.setOnCheckedChangeListener { buttonView, isChecked ->
            if (isChecked) {
                intent.action = "on"
            } else {
                intent.action = "off"
            }

            startService(intent)
        }
    }
}
